src/components/labelPrinting/LabelPrintingForm.vue

```diff
@@ -112,114 +95,74 @@
+  computed: {
+    printerOptions() {
+      return this.$store.getters.printers.map((name) => ({
         text: name,
       }))
-      this.printerOptions = printerOptions
-    },
-    suffixedBarcodes() {
-      let listSuffixedBarcodes = []
-      let noOfLabels = this.form.selectedNumberOfLabels
-      let applyLabels = noOfLabels > 1 && noOfLabels <= 9
-
-      //Append the four letter suffix and/or the label number, if given to the barcodes
-      if (this.form.barcode && this.form.selectedSuffix) {
-        let barcodes = this.form.barcode.split(/\r?\n|\r|\n/g)
-        switch (true) {
-          case this.suffix() && applyLabels:
-            //Add both suffix and label number to the barcodes to be printed
-            listSuffixedBarcodes = this.appendSuffixWithLabels(
-              barcodes,
-              listSuffixedBarcodes,
-              noOfLabels,
-            )
-            break
-          case this.suffix() && !applyLabels:
-            //Add only suffix to the barcodes to be printed
-            listSuffixedBarcodes = this.appendSuffix(barcodes, listSuffixedBarcodes)
-            break
-          case applyLabels && !this.suffix():
-            //Add only label number to the barcodes to be printed
-            listSuffixedBarcodes = this.appendLabels(barcodes, listSuffixedBarcodes, noOfLabels)
-            break
-          case !applyLabels && !this.suffix():
-            //Just the plain barcodes list to be printed
-            for (let barcode of barcodes) {
-              listSuffixedBarcodes.push(barcode)
-            }
-            break
-        }
-      }
-      return listSuffixedBarcodes
     },
-    appendSuffix(barcodes, listSuffixedBarcodes) {
-      for (let barcode of barcodes) {
-        listSuffixedBarcodes.push(barcode.concat('-', this.suffix()))
-      }
-      return listSuffixedBarcodes
+    suffixOptions() {
+      return createSuffixDropdownOptions(SuffixList)
     },
-    appendLabels(barcodes, listSuffixedBarcodes, noOfLabels) {
-      for (let barcode of barcodes) {
-        for (let i = 0; i < noOfLabels; i++) {
-          listSuffixedBarcodes.push(barcode.concat('-', i + 1))
-        }
-      }
-      return listSuffixedBarcodes
+    suffixItems() {
+      return createSuffixItems(SuffixList)
     },
-    appendSuffixWithLabels(barcodes, listSuffixedBarcodes, noOfLabels) {
-      for (let barcode of barcodes) {
-        for (let i = 0; i < noOfLabels; i++) {
-          listSuffixedBarcodes.push(barcode.concat('-', this.suffix(), '-', i + 1))
-        }
-      }
-      return listSuffixedBarcodes
-    },
-    suffix() {
-      return this.form.selectedSuffix === 'No suffix' ? '' : this.form.selectedSuffix.slice(0, 4)
-    },
-    printerName() {
-      return this.form.selectedPrinterName
+    labels() {
+      const date = getCurrentDate()
+      const suffixItem = this.suffixItems[this.form.suffix]
+
+      // it is possible for there to be no barcodes so we need to add a guard
```

Comment:
Maybe add a test for this guard

Reply:
The current tests already cover it. Because it is reactive any test that does not have barcodes was failing which is why I added a guard. So the tests started failing and then I added the guard. Hope that makes sense?